server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -297,6 +298,29 @@ public JobDetail hypervisorUpdateAsync(
         return HypervisorUpdateJob.forOwner(owner, hypervisorJson, createMissing, principal, reporterId);
     }
 
+    @ApiOperation(notes = "Updates last check in date of all consumers of the given reporterId.",
+        value = "hypervisorHeartbeatUpdate")
+    @ApiResponses({
+        @ApiResponse(code = 202, message = ""),
+        @ApiResponse(code = 400, message = "Illegal reporter ID was provided"),
+        @ApiResponse(code = 404, message = "Target owner not found.")})
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Transactional
+    @Path("/heartbeat/{owner}")
```

Comment:
Something I did not think about before is the design of the API itself.

- For the verb: I would say that this should be either a PATCH or a PUT instead of a POST (technically, it fits much more to use a PATCH, since we don't accept an entire entity, like a `ConsumerDTO` or a `HypervisorDTO` to be updated, but we only want to update a specific field of the entity, the last check in date).

- For the url: I think the `{owner}` should be at the start, since it is the top-level entity, which is consistent with all other endpoints we have. Also, maybe the `reporter_id` should also be in the URL, since this is what identifies the entity/ies being PATCHed. So the url should be one of: `/{owner}/heartbeat`,  `/{owner}/{reporter_id}/heartbeat` or `/{owner}/heartbeat/{reporter_id}`.

@wottop @Ceiu @Januson What are your thoughts?

Reply:
My opinion is that it stays on the HypervisorResource. It mirrors the existing async endpoint for the hypervisor check in in the url and use of reporter id as a parameter.

Reply:
/{owner}/heartbeat assumes that there is only ever one type of thing that we are reporting a heartbeat for. Does it need to be scoped by the type of heartbeat?

Reply:
Done.